src/ringcentral-web-phone.js

```diff
@@ -1155,6 +1167,242 @@
+            timestamp: {
+                start: '',
+                stop: ''
+            },
+
+            netType: '',
+
+            jitterArr:[],
+
+            packetLost :0,
+            packetsReceived: 0,
+            jitterBufferNominal: 0,
+            jitterBufferMax: 0,
+            networkPacketLossRate: 0,
+            jitterBufferDiscardRate: 0,
+            totalSumJitter:0,
+            totalIntervalCount:-1,
+
+            NLR : 0,
+            JBM : 0 ,
+            JBN :  0,
+            JDR : 0,
+            MOSLQ: 0
+        }
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+    /**
+     * @this {SIP.Session}
+     */
+    //TODO: Remove options
+    function getQOSStats(options){
+        var session =  this;
+        var peer =  session.sessionDescriptionHandler.peerConnection;
+        
+        session.qosStatsObj.callID = session.request.call_id||'';
+        session.qosStatsObj.fromTag  = session.from_tag || '';
+        session.qosStatsObj.toTag  = session.to_tag || '';
+        session.qosStatsObj.localID = session.request.headers.From[0].raw||session.request.headers.From[0];
+        session.qosStatsObj.remoteID = session.request.headers.To[0].raw||session.request.headers.To[0];
+        session.qosStatsObj.origID = session.request.headers.From[0].raw||session.request.headers.From[0];
+
+        this.options = options || {};
+        getStat(peer, session);
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+
+    function average(array) {
+        var sum = array.reduce((a, b) => a + b, 0);
+        var avg = (sum / array.length);
+        return avg;
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+    function addToMap(map, key, value) {
+        if (key in map) {
+            map[key] =  parseInt(map[key],10) + 1;
+        } else {
+            map[key] = parseInt(1);
+        }
+        return map;
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+    function getNetworkType(connectionType){
+        var sysNetwork = connectionType.systemNetworkType || 'unknown';
+        var localNetwork = connectionType.local.networkType || ['unknown'];
+
+        if (!sysNetwork || sysNetwork === 'unknown') {
+            return localNetwork[0];
+        }
+        return sysNetwork;
+    }
+
+    /*--------------------------------------------------------------------------------------------------------------------*/
+
+    //TODO: Check/test networktype values
+
+
+    function getStat(peer, session){
+       
+        var repeatInterval = session.ua.qosCollectInterval;
+        var qosStatsObj = session.qosStatsObj;
+
+        getStats(peer, function (getStatsResult){
```

Comment:
You need too add this dependency through UMD declaration that we use for WebPhone (in the beginning of the file)